lib/ansible/plugins/action/copy.py

```diff
@@ -310,13 +310,11 @@ def _copy_file(self, source_full, source_rel, content, content_tempfile,
             self._remove_tempfile_if_content_defined(content, content_tempfile)
             self._loader.cleanup_tmp_file(source_full)
 
-            # FIXME: I don't think this is needed when PIPELINING=0 because the source is created
-            # world readable.  Access to the directory itself is controlled via fixup_perms2() as
-            # part of executing the module. Check that umask with scp/sftp/piped doesn't cause
-            # a problem before acting on this idea. (This idea would save a round-trip)
-            # fix file permissions when the copy is done as a different user
+            # File source permissions are attempted to be preserved via sftp/scp or local
+            # but they might fail and/or use other connections or transfer methods
+            # So we ensure that the remote is readable by remote user (could be moved to copy module)
             if remote_path:
-                self._fixup_perms2((self._connection._shell.tmpdir, remote_path))
+                self._fixup_perms2((self._connection._shell.tmpdir, remote_path), execute=False)
```

Comment:
It looks like these changes are unrelated to fact gathering. Were they supposed to be on another PR?

Reply:
yep